Objects/unicodeobject.c

```diff
@@ -2217,6 +2217,19 @@ PyUnicode_FromWideChar(const wchar_t *u, Py_ssize_t size)
     if (size == 0)
         _Py_RETURN_UNICODE_EMPTY();
 
+#ifdef HAVE_NON_UNICODE_WCHAR_T_REPRESENTATION
+    /* Oracle Solaris uses non-Unicode internal wchar_t form for
+       non-Unicode locales and hence needs conversion to UTF first. */
+    if (!_Py_IsLocaleUnicodeWchar()) {
+        char32_t* c32 = _Py_convert_wchar_t_to_UCS4(u, size);
+        if (!c32)
+            return NULL;
```

Comment:
PEP 7: please add braces { ... } for new code.

Reply:
Resolved on all places.